packages/astro/src/core/dev/index.ts

```diff
@@ -92,49 +94,94 @@ export class AstroDevServer {
   }
 
   public async handleHotUpdate({ file, modules }: HmrContext): Promise<void | ModuleNode[]> {
-    if (!this.viteServer) throw new Error(`AstroDevServer.start() not called`);
+    const { viteServer } = this;
+    if (!viteServer) throw new Error(`AstroDevServer.start() not called`);
```

Comment:
Only referenced a property on `this` here because within a function below TypeScript forgot we checked for it (which is technically correct; executes in different scope)